### ISSUE TITLE
UniversalDecoder can't decode Job config file 

### ISSUE THREAD

**Issue by kenontech** [kubernetes/client-go#277]
Hi there,

I was trying to use the UniversalDecoder to decode a Job yaml config file, but go an error:
```
import "k8s.io/client-go/pkg/api"
...
obj, kind, err := api.Codecs.UniversalDecoder.Decode([]byte(doc), nil, nil)
...
Error: could not decode yaml:
...
no kind "Job" is registered for version "batch/v1"
```
The branch I am using is v4.0.0

Is there any special consideration not supporting Job type resources in UniversalDecoder?

**Comment by dpetersen** [comment]
I had the exact same problem you had, with code that looked identical to what you've provided above. It worked in a previous version of this client, then stopped working on my attempted upgrade to 4.0. I've made some modifications to my code that appear to fix the issue and make everything work like it did before. Previously I had the Decode line identical to yours, and now I have:

```go
import (
  "k8s.io/apimachinery/pkg/runtime/serializer"
  clientsetscheme "k8s.io/client-go/kubernetes/scheme"
)
...
deserializer := serializer.NewCodecFactory(clientsetscheme.Scheme).UniversalDeserializer()
obj, _, err := deserializer.Decode(bs, nil, nil)
```

The `clientsetscheme` name and idea I got from docs in [some registration code](https://github.com/kubernetes/client-go/blob/release-4.0/kubernetes/scheme/register.go#L55-L68), and the `serializer.NewCodecFactory` comes from [the code](https://github.com/kubernetes/client-go/blob/release-4.0/pkg/api/register.go#L45) that was actually getting called when we previously used `api.Codecs`. 

Maybe this isn't the right way to do it, but it does seem to work. I am now properly parsing a Job definition from YAML and using it in the same way that I was before. I hope it works for you, too!
